File: admin/src/main/java/com/project/admin/service/dto/techblog/TechBlogResult.java
package com.project.admin.service.dto.techblog;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.project.admin.constant.BlogRegion;
import com.project.admin.constant.Source;
import com.project.admin.domain.entity.TechBlog;
import java.time.LocalDateTime;

public record TechBlogResult(
        Long id,
        Source source,
        BlogRegion blogRegion,
        String title,
        String url,
        @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss", timezone = "Asia/Seoul")
        LocalDateTime publishedAt) {

    public static TechBlogResult from(TechBlog techBlog) {
        return new TechBlogResult(
                techBlog.getId(),
                techBlog.getSource(),
                techBlog.getBlogRegion(),
                techBlog.getTitle(),
                techBlog.getUrl(),
                techBlog.getPublishedAt()
        );
    }
}

File: api/src/main/kotlin/com/project/api/service/dto/TechBlogResult.kt
package com.project.api.service.dto

import com.project.api.constants.Region
import com.project.api.constants.Source
import com.project.api.entity.TechBlog
import java.time.LocalDateTime

data class TechBlogResults(
    val techBlogs: List<TechBlogResult>,
    val pagination: PageResult,
) {
    companion object {
        fun empty(page: Int, size: Int) = TechBlogResults(
            techBlogs = emptyList(),
            pagination = PageResult(totalCount = 0, pageSize = size, currentPage = page),
        )
    }
}

data class TechBlogResult(
    val id: Long,
    val source: Source,
    val region: Region,
    val title: String,
    val publishedAt: LocalDateTime,
    val tags: List<String>,
) {
    companion object {
        fun from(techBlog: TechBlog) = TechBlogResult(
            id = techBlog.id,
            source = techBlog.source,
            region = techBlog.region,
            title = techBlog.title,
            publishedAt = techBlog.publishedAt,
            tags = techBlog.tags ?: emptyList(),
        )
    }
}
